var/spack/repos/builtin/packages/openmpi/package.py

```diff
@@ -142,63 +143,56 @@ def verbs(self):
         elif self.spec.satisfies('@1.7:'):
             return 'verbs'
 
-    def install(self, spec, prefix):
+    @AutotoolsPackage.precondition('autoreconf')
+    def die_without_fortran(self):
         # Until we can pass variants such as +fortran through virtual
         # dependencies depends_on('mpi'), require Fortran compiler to
         # avoid delayed build errors in dependents.
         if (self.compiler.f77 is None) or (self.compiler.fc is None):
-            raise InstallError('OpenMPI requires both C and Fortran ',
-                               'compilers!')
-
-        config_args = ["--prefix=%s" % prefix,
-                       "--with-hwloc=%s" % spec['hwloc'].prefix,
-                       "--enable-shared",
-                       "--enable-static"]
+            raise InstallError(
+                'OpenMPI requires both C and Fortran compilers!'
+            )
 
-        # for Open-MPI 2.0:, C++ bindings are disabled by default.
-        if self.spec.satisfies('@2.0:'):
-            config_args.extend(['--enable-mpi-cxx'])
+    def configure_args(self):
+        spec = self.spec
 
-        if getattr(self, 'config_extra', None) is not None:
-            config_args.extend(self.config_extra)
-
-        # Variant based arguments
-        config_args.extend([
+        config_args = [
+            '--enable-shared',
+            '--enable-static'
+            '--with-hwloc={0}'.format(spec['hwloc'].prefix),
             # Schedulers
             '--with-tm' if '+tm' in spec else '--without-tm',
             '--with-slurm' if '+slurm' in spec else '--without-slurm',
             # Fabrics
             '--with-psm' if '+psm' in spec else '--without-psm',
             '--with-psm2' if '+psm2' in spec else '--without-psm2',
+            '--with-pmi' if '+pmi' in spec else '--without-pmi',
             '--with-mxm' if '+mxm' in spec else '--without-mxm',
             # Other options
-            ('--enable-mpi-thread-multiple' if '+thread_multiple' in spec
-                else '--disable-mpi-thread-multiple'),
-            '--with-pmi' if '+pmi' in spec else '--without-pmi',
+            '--enable-java' if '+java' in spec else '--disable-java',
+            '--enable-mpi-java' if '+java' in spec else '--disable-mpi-java',
             '--with-sqlite3' if '+sqlite3' in spec else '--without-sqlite3',
             '--enable-vt' if '+vt' in spec else '--disable-vt'
```

Comment:
Good catch. Sometimes Python is a little too loose for my taste, because everything installed fine with no complaints. Although now that I'm rechecking my build output, I'm seeing that many of these options are deprecated now. Going to investigate further.